RocketChat/Rocket.Chat.ReactNative
# fix: fix bugs related to auto-translate and add tests

## Proposed changes

The task was initially related to the bug reported in this task: [SUP-240]
However, looking deeper into the task, several bugs were found in the auto translate functionality and to correct these bugs this PR was created.

- Fix bug of enabling auto translate and messages not being updated after activation/deactivation.
- Fix bug with re-rendering in room view when feature is turned on or off.
- Fixes the error of attachments not considering translations.
- Adds tests to auto translate functionality.

## Issue(s)	


## How to test or reproduce


- Enable auto translate in a room and send a message by one user, enter that room with another user and watch old conversations get translated.
- Send messages in real time and see if they are translated.
- Send attachments and verify that the attachment message is being translated.
- Check that the view original message action is working correctly. (hold the message and select the option in the action sheet)
- Check if by disabling auto translate the messages will display the original content again.
- Check if when disabling auto translate the option to show the translation will be unavailable.
- Check if when enabling auto translate again the selected language will be the previously selected language.

## Screenshots

## Types of changes




## Checklist



## Further comments


Jira: [SUP-240]

[SUP-240]: https://rocketchat.atlassian.net/browse/SUP-240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SUP-240]: https://rocketchat.atlassian.net/browse/SUP-240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
